.github/actions/build-action/action.yml

```diff
@@ -0,0 +1,162 @@
+    description: 'Options to use in tsan builds'
+    required: false
+  NINJA_BUILD:
+    description: 'Boolean: Build with Ninja'
+    required: false
+  CMAKE_EXE:
+    description: 'Use a specific instance of CMake'
+    required: false
+outputs:
+  random-number:
+    description: "Random number"
+    value: ${{ steps.random-number-generator.outputs.random-id }}
+runs:
+  using: "composite"
+  steps:
+    - name: Dump-environment
+      shell: bash
+      run: |
+        env
+        set | grep =
+
+    - name: Install ccache
+      shell: bash
+      run: |
+        if ! type -a ccache
+        then
+          apt install ccache
+        fi
+
+    - name: Reset ccache stats
+      shell: bash
+      run: |
+          ccache -s
+          # Reset the stats so only this build counts
+          ccache -z
+
+    - name: Disk space
+      shell: bash
+      run: |
+        df -h
+        if [[ -d "${CACHE_DIR}" ]]
+        then
+            du -sh ${CACHE_DIR}
+        fi
+        if [[ -d "${CCACHE_DIR}" ]]
+        then
+            du -sh ${CCACHE_DIR}
+        fi
+        if [[ -d "${NIH_CACHE_ROOT}" ]]
+        then
+            find ${NIH_CACHE_ROOT} -maxdepth 3 \( -iname src -prune -o \
+              -type d -exec du -sh {} \; \)
+            # Post-condition: The cache will not be reused to build with a
+            # different compiler
+            if [[ "${CC}" =~ "gcc" ]]
+            then
+              echo Erasing NIH cache for clang
+              rm -rf "${NIH_CACHE_ROOT}"/*/Clang_*/ || true
+            elif [[ "${CC}" =~ "clang" ]]
+            then
+              echo Erasing NIH cache for gcc
+              rm -rf "${NIH_CACHE_ROOT}"/*/GNU_*/ || true
+            fi
+            # Free up space if the cache is being reused with a different
+            # generator from prior builds (ninja vs. makefiles).
+            # Post condition: Cache will not be reused with a different
+            # generator.
+            if [[ "${NINJA_BUILD}" == "true" && -d "${NIH_CACHE_ROOT}"/unix_makefiles && \
+              $( du -sm "${NIH_CACHE_ROOT}"/unix_makefiles | cut -f1 ) -gt 1000  ]]
+            then
+              echo Erasing NIH cache for makefiles
+              rm -rf "${NIH_CACHE_ROOT}"/unix_makefiles || true
+            elif [[ "${NINJA_BUILD}" == "false" && -d "${NIH_CACHE_ROOT}"/ninja && \
+              $( du -sm "${NIH_CACHE_ROOT}"/ninja | cut -f1 ) -gt 1000  ]]
+            then
+              echo Erasing NIH cache for ninja
+              rm -rf "${NIH_CACHE_ROOT}"/ninja || true
+            fi
+        fi
+
+    - name: Build and test
+      shell: bash
+      run: |
+          export CXX=${CC/gcc/g++}
+          export CXX=${CXX/clang/clang++}
+          export NUM_PROCESSORS=${NUM_PROCESSORS:-$( nproc )}
+          echo NUM_PROCESSORS is ${NUM_PROCESSORS}
+          if [ "${{ inputs.image }}" == "rippled-ci-builder:2944b78d22db" ]
+          then
+              export CMAKE_EXTRA_ARGS+=" -DBoost_NO_BOOST_CMAKE=ON"
```

Comment:
I wanted to only need to check it in one (or two) place(s), instead of every time the image is specified. Unfortunately, the workflow specification doesn't really have any kind of conditional operations, so I can't write (for example):
```
      - name: Build and test
        uses: ./.github/actions/build-action
        with:
          no_boost_cmake: ${{ matrix.image == "rippled-ci-builder:2944b78d22db" ? "ON" : "OFF" }}
```
Requiring it to be specified in the same place as the image can get complicated, especially in the matrix.

Plus, I figure when we inevitably get a new image, I can update the check as appropriate.

Reply:
As mentioned above, I updated the matrix to use composite values and no longer require checking `image`.